### PR TITLE
feat: add lock event API

### PR DESCRIPTION
**Pull request by purplesmoke05**
Close: part of #1286 

### 1. Storing data of locked balances

- New TBL
  - `lock`
  - `unlock`
- Fix indexer batch
  - indexer_Position_Bond.py
  - indexer_Position_Share.py

### 2. Add locked balances to the return item of Position API

List and individual references of token holders
  - GET: /Token/{token_address}/Holders
  - GET: /Token/{token_address}/Holders/Collection/{list_id}

### 3. Add transfer indexing and holders collection creation process to watch for unlock events

Add transfer history when unlock event occurs
  - indexer_Transfer.py

### 4. List all account's locked positions

  - [NEW] GET: /Position/{account_address}/Share/Lock
  - [NEW] GET: /Position/{account_address}/StraightBond/Lock

In addition, list lock and unlock events

  - [NEW] GET: /Position/{account_address}/Share/Lock/Event
  - [NEW] GET: /Position/{account_address}/StraightBond/Lock/Event

### CODE REVIEW

File: batch/indexer_Position_Share.py [hunk 1]
```diff
@@ -357,6 +360,22 @@ def __sync_lock(self, db_session: Session, block_to: int):
...
+                        db_session=db_session,
+                        transaction_hash=event["transactionHash"].hex(),
+                        block_number=event["blockNumber"],
+                        token_address=token.address,
+                        lock_address=lock_address,
+                        account_address=account_address,
+                        value=value,
+                        data_str=data,
+                        block_timestamp=event_created
+                    )
```
Comment (purplesmoke05) [inline batch/indexer_Position_Share.py:378]: Added logic for storing lock/unlock data.

File: batch/indexer_Position_Bond.py [hunk 2]
```diff
@@ -357,6 +360,22 @@ def __sync_lock(self, db_session: Session, block_to: int):
...
+                        db_session=db_session,
+                        transaction_hash=event["transactionHash"].hex(),
+                        block_number=event["blockNumber"],
+                        token_address=token.address,
+                        lock_address=lock_address,
+                        account_address=account_address,
+                        value=value,
+                        data_str=data,
+                        block_timestamp=event_created
+                    )
```
Comment (purplesmoke05) [inline batch/indexer_Position_Bond.py:378]: Added logic for storing lock/unlock data.

File: app/model/schema/token.py [hunk 3]
```diff
@@ -109,7 +107,8 @@ class CreateTokenHoldersCollectionResponse(BaseModel):
 class TokenHoldersCollectionHolder(BaseModel):
     account_address: str = Field(description="Account address of token holder.")
     hold_balance: int = Field(description="Amount of balance."
-                                          "This includes balance/pending_transfer/exchange_balance/exchange_commitment.")
+                                          "This includes balance/pending_transfer/exchange_balance/exchange_commitment/locked.")
+    current_locked_balance: int = Field(description="Amount of locked in current.")
```
Comment (purplesmoke05) [inline app/model/schema/token.py:111]: Added an attribute which indicates current locked balances.

File: app/api/routers/token.py [hunk 4]
```diff
@@ -424,11 +423,32 @@ def get_token_holders_collection(
...
         all()
-    token_holders = [_token_holder.json() for _token_holder in _token_holders]
+    )
+    token_holders = [
+        {
+            **_token_holder[0].json(),
+            "current_locked_balance": _token_holder[1] or 0
+        }
+        for _token_holder in _token_holders
+    ]
```
Comment (purplesmoke05) [inline app/api/routers/token.py:451]: Added logic for merging current locked balance to holder collection.

File: app/api/routers/position.py [hunk 5]
```diff
@@ -82,19 +94,297 @@
...
+    response_model=GenericSuccessResponse[LockEventsResponse],
+    responses=get_routers_responses()
+)
+def list_all_straight_bond_lock_events(
+    data: dict = Depends(ListAllLockEvent(config.BOND_TOKEN_ENABLED))
+):
+    return json_response({
+        **SuccessResponse.default(),
+        "data": data
+    })
```
Comment (purplesmoke05) [inline app/api/routers/position.py:385]: Added API endpoints for "lock state" and "lock event".

**Review by YoshihitoAso — APPROVED** [review]

**Pull request merged**